diff --git a/core/modules/layout_builder/src/EventSubscriber/BlockComponentRenderArray.php b/core/modules/layout_builder/src/EventSubscriber/BlockComponentRenderArray.php
index e1a9773959..fee6573cc1 100644
--- a/core/modules/layout_builder/src/EventSubscriber/BlockComponentRenderArray.php
+++ b/core/modules/layout_builder/src/EventSubscriber/BlockComponentRenderArray.php
@@ -102,7 +102,7 @@ public function onBuildRender(SectionComponentBuildRenderArrayEvent $event) {
       }
 
       $content = $block->build();
-      $is_content_empty = Element::isEmpty($content);
+      $is_content_empty = !$content || Element::isEmpty($content) || ($event->inPreview() && $this->containsForm($content));
       $is_placeholder_ready = $event->inPreview() && $block instanceof PreviewFallbackInterface;
       // If the content is empty and no placeholder is available, return.
       if ($is_content_empty && !$is_placeholder_ready) {
@@ -117,7 +117,7 @@ public function onBuildRender(SectionComponentBuildRenderArrayEvent $event) {
         '#base_plugin_id' => $block->getBaseId(),
         '#derivative_plugin_id' => $block->getDerivativeId(),
         '#weight' => $event->getComponent()->getWeight(),
-        'content' => $content,
+        'content' => !$is_content_empty ? $content : [],
       ];
 
       if ($block instanceof PreviewFallbackInterface) {
@@ -141,4 +141,19 @@ public function onBuildRender(SectionComponentBuildRenderArrayEvent $event) {
     }
   }
 
+  /**
+   */
+  protected function containsForm(array $content) {
+    if (isset($content['#type']) && ($content['#type'] === 'form' || $content['#type'] === 'webform')) {
+      return TRUE;
+    }
+
+    foreach (Element::children($content) as $key) {
+      if ($this->containsForm($content[$key])) {
+        return TRUE;
+      }
+    }
+    return FALSE;
+  }
+
 }