diff --git a/core/modules/layout_builder/src/Entity/LayoutBuilderEntityViewDisplay.php b/core/modules/layout_builder/src/Entity/LayoutBuilderEntityViewDisplay.php
index 78ff430b..f5a08c58 100644
--- a/core/modules/layout_builder/src/Entity/LayoutBuilderEntityViewDisplay.php
+++ b/core/modules/layout_builder/src/Entity/LayoutBuilderEntityViewDisplay.php
@@ -220,6 +220,9 @@ public function buildMultiple(array $entities) {
           '@entity' => $entity->getEntityType()->getSingularLabel(),
         ]);
         $contexts['layout_builder.entity'] = EntityContext::fromEntity($entity, $label);
+
+        \Drupal::moduleHandler()->alter('layout_builder_view_context', $contexts);
+
         foreach ($sections as $delta => $section) {
           $build_list[$id]['_layout_builder'][$delta] = $section->toRenderArray($contexts);
         }
diff --git a/core/modules/layout_builder/src/Plugin/SectionStorage/DefaultsSectionStorage.php b/core/modules/layout_builder/src/Plugin/SectionStorage/DefaultsSectionStorage.php
index 6fab1199..00639095 100644
--- a/core/modules/layout_builder/src/Plugin/SectionStorage/DefaultsSectionStorage.php
+++ b/core/modules/layout_builder/src/Plugin/SectionStorage/DefaultsSectionStorage.php
@@ -258,6 +258,10 @@ public function getContexts() {
 
     $contexts = [];
     $contexts['layout_builder.entity'] = EntityContext::fromEntity($entity);
+
+    $allow_sample = TRUE;
+    \Drupal::moduleHandler()->alter('layout_builder_view_context', $contexts, $allow_sample);
+
     return $contexts;
   }
 
diff --git a/core/modules/layout_builder/src/Plugin/SectionStorage/OverridesSectionStorage.php b/core/modules/layout_builder/src/Plugin/SectionStorage/OverridesSectionStorage.php
index 3a92ee63..fca51bcf 100644
--- a/core/modules/layout_builder/src/Plugin/SectionStorage/OverridesSectionStorage.php
+++ b/core/modules/layout_builder/src/Plugin/SectionStorage/OverridesSectionStorage.php
@@ -214,6 +214,9 @@ public function getLayoutBuilderUrl($rel = 'view') {
   public function getContexts() {
     $entity = $this->getEntity();
     $contexts['layout_builder.entity'] = EntityContext::fromEntity($entity);
+
+    $allow_sample = TRUE;
+    \Drupal::moduleHandler()->alter('layout_builder_view_context', $contexts, $allow_sample);
     return $contexts;
   }
 
