diff --git a/js/termData.js b/js/termData.js
index 4b1357c..18edbad 100644
--- a/js/termData.js
+++ b/js/termData.js
@@ -3,7 +3,7 @@
 
   Drupal.TaxonomyManagerTermData = function (tid, tree) {
     // We change the hidden form element which then triggers the AJAX system.
-    $('input[name=load-term-data]').val(tid).trigger('change');
+    window.location.href = '/taxonomy/term/' + tid + '/edit' + '?destination=' + window.location.pathname;
   };
 
 })(jQuery, Drupal, drupalSettings);
diff --git a/js/tree.js b/js/tree.js
index d8e4e1c..203af15 100644
--- a/js/tree.js
+++ b/js/tree.js
@@ -71,8 +71,10 @@
           }
         }
       },
-      focus: function (event, data) {
-        new Drupal.TaxonomyManagerTermData(data.node.key, data.tree);
+      click: function (event, data) {
+        if(data.targetType === 'title'){
+          new Drupal.TaxonomyManagerTermData(data.node.key, data.tree);
+        }
       }
     });
   };
