diff --git a/library_manager.module b/library_manager.module
index 07ee69a..f6f59cb 100644
--- a/library_manager.module
+++ b/library_manager.module
@@ -176,6 +176,7 @@ function library_manager_page_attachments(array &$attachments) {
   foreach ($definitions as $definition) {
     foreach ($definition->getVisibilityConditions() as $condition) {
 
+      $attach = TRUE;
       $missing_context = FALSE;
       if ($condition instanceof ContextAwarePluginInterface) {
         try {
@@ -188,11 +189,14 @@ function library_manager_page_attachments(array &$attachments) {
       }
 
       if ($missing_context || !$condition->execute()) {
-        break 2;
+        $attach = FALSE;
+        break;
       }
 
     }
-    $attachments['#attached']['library'][] = 'library_manager/' . $definition->id();
+    if ($attach) {
+      $attachments['#attached']['library'][] = 'library_manager/' . $definition->id();
+    }
   }
 }
 
