diff --git a/core/modules/book/src/BookManager.php b/core/modules/book/src/BookManager.php
index ea9c1fa5d..a2009ed07 100644
--- a/core/modules/book/src/BookManager.php
+++ b/core/modules/book/src/BookManager.php
@@ -626,11 +626,12 @@ protected function buildItems(array $tree) {
         $element['in_active_trail'] = TRUE;
       }
 
+      $language = \Drupal::languageManager()->getCurrentLanguage();
       // Allow book-specific theme overrides.
       $element['attributes'] = new Attribute();
       $element['title'] = $data['link']['title'];
       $node = $this->entityTypeManager->getStorage('node')->load($data['link']['nid']);
-      $element['url'] = $node->toUrl();
+      $element['url'] = $node->toUrl('canonical', ['language' => $language]);
       $element['localized_options'] = !empty($data['link']['localized_options']) ? $data['link']['localized_options'] : [];
       $element['localized_options']['set_active_class'] = TRUE;
       $element['below'] = $data['below'] ? $this->buildItems($data['below']) : [];
