diff --git a/js/libs/jquery.simpler-select.js b/js/libs/jquery.simpler-select.js
index 52be7dc..53d0cf2 100755
--- a/js/libs/jquery.simpler-select.js
+++ b/js/libs/jquery.simpler-select.js
@@ -113,6 +113,11 @@

       var that = this;
       var $select = $('<select class="simpler-select">').addClass(that.$element.attr('class'));
+
+      if(that.$element.attr('disabled')){
+        $select.attr('disabled',that.$element.attr('disabled'))
+      }
+
       var $wrapper = $('<div class="select-wrapper">');

       if (that.$element.hasClass('error')) {
