diff --git a/src/Plugin/Field/FieldWidget/CshsWidget.php b/src/Plugin/Field/FieldWidget/CshsWidget.php
index 71e1f1b..d374a67 100644
--- a/src/Plugin/Field/FieldWidget/CshsWidget.php
+++ b/src/Plugin/Field/FieldWidget/CshsWidget.php
@@ -56,7 +56,7 @@ class CshsWidget extends WidgetBase {
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
     $element['target_id'] = array_merge($element, $this->helperFormElement(), [
       '#label' => $this->fieldDefinition->getLabel(),
-      '#default_value' => $items->get($delta)->getValue(),
+      '#default_value' => isset($items[$delta]->target_id) ? $items[$delta]->target_id : [],
     ]);
 
     return $element;
