diff --git a/src/Service/ViewsBulkOperationsActionProcessor.php b/src/Service/ViewsBulkOperationsActionProcessor.php
index 481ee4c..9ecc1f9 100644
--- a/src/Service/ViewsBulkOperationsActionProcessor.php
+++ b/src/Service/ViewsBulkOperationsActionProcessor.php
@@ -210,6 +210,26 @@ class ViewsBulkOperationsActionProcessor implements ViewsBulkOperationsActionPro
       $this->view->setExposedInput(['_views_bulk_operations_override' => TRUE]);
     }
 
+    // In some cases we may encounter nondeterministic bahaviour in
+    // db queries with sorts allowing different order of results.
+    // To fix this we're removing all sorts and setting one sorting
+    // rule by the view base id field.
+    $sorts = $this->view->getHandlers('sort');
+    foreach ($sorts as $id => $sort) {
+      $this->view->setHandler($this->bulkFormData['display_id'], 'sort', $id, NULL);
+    }
+    $base_field = $this->view->storage->get('base_field');
+    $this->view->setHandler($this->bulkFormData['display_id'], 'sort', $base_field, [
+      'id' => $base_field,
+      'table' => $this->view->storage->get('base_table'),
+      'field' => $base_field,
+      'order' => 'ASC',
+      'relationship' => 'none',
+      'group_type' => 'group',
+      'exposed' => 'FALSE',
+      'plugin_id' => 'standard',
+    ]);
+
     $this->view->setItemsPerPage($this->bulkFormData['batch_size']);
     $this->view->setCurrentPage($page);
     $this->view->build();
